linode/networkingip/framework_resource_schema.go

```diff
@@ -0,0 +1,68 @@
+package networkingip
+
+import (
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/boolplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+)
+
+var frameworkResourceSchema = schema.Schema{
+	Attributes: map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Description: "The ID of the IPv4 address.",
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"linode_id": schema.Int64Attribute{
+			Description: "The ID of the Linode to allocate an IPv4 address for. Required when reserved is false or not set.",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.UseStateForUnknown(),
+			},
+		},
+		"reserved": schema.BoolAttribute{
+			Description: "Whether the IPv4 address should be reserved.",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"region": schema.StringAttribute{
+			Description: "The region for the reserved IPv4 address. Required when reserved is true and linode_id is not set.",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"public": schema.BoolAttribute{
+			Description: "Whether the IPv4 address is public or private.",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"address": schema.StringAttribute{
+			Description: "The allocated IPv4 address.",
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"type": schema.StringAttribute{
+			Description: "The type of IP address (ipv4).",
+			Optional:    true,
```

Comment:
Could this be made either required or be given a default value of IPv4?

Reply:
Maybe not required, but having a default value would definitely be useful to this resource